File: assignment-2/part1/section2/bank_account.py
class BankAccount:
    def __init__(self, user, balance):
        print(f"Creating account for user {user} with ${balance} amount.")
        self.user = user
        self._balance = balance

    def deposit(self, amount):
        print(f"Adding ${amount} to your account.")
        self._balance += amount

    def withdraw(self, amount):
        print(f"Trying to withdraw ${amount} amount.")
        if amount <= self._balance:
            print("Valid withdrawal amount!")
            self._balance -= amount
        else:
            print("Insufficient funds.")

    def display_balance(self):
        print(f"{self.user}'s balance: ${self._balance}.")


account = BankAccount("Furqaan", 1000)
account.deposit(500)
account.withdraw(300)
account.display_balance()

File: assignment-2/part1/section2/bank_account.cpp
#include <iostream>
#include <string>
using namespace std;

class BankAccount {
private:
    string user;
    double balance;

public:
    BankAccount(string accountUser, double initialBalance) {
        cout << "Creating account for user " << accountUser << " with $" << initialBalance << " amount." << endl;
        user = accountUser;
        balance = initialBalance;
    }

    void deposit(double amount) {
        cout << "Adding $" << amount << " to your account." << endl;
        balance += amount;
    }

    void withdraw(double amount) {
        cout << "Trying to withdraw $" << amount << " amount." << endl;
        if (amount <= balance) {
            cout << "Valid withdrawal amount!" << endl;
            balance -= amount;
        } else {
            cout << "Insufficient funds." << endl;
        }
    }

    void displayBalance() {
        cout << user << "'s balance: $" << balance << "." << endl;
    }
};

int main() {
    BankAccount account("Furqaan", 1000);
    account.deposit(500);
    account.withdraw(300);
    account.displayBalance();

    return 0;
}
